cmd/logger.go

```diff
@@ -137,7 +111,38 @@ func getSource() string {
 	return fmt.Sprintf("[%s:%d:%s()]", filename, lineNum, funcName)
 }
 
-func logIf(level logrus.Level, source string, err error, msg string, data ...interface{}) {
+// getTrace method - creates and returns stack trace
+func getTrace(traceLevel int) []string {
+	var trace []string
+	pc, file, lineNumber, ok := runtime.Caller(traceLevel)
+	rootPath := filepath.Dir(filepath.Dir(file)) + string(os.PathSeparator)
+	for ok {
+		_, funcName := filepath.Split(runtime.FuncForPC(pc).Name())
+		// Clean up the common root path, 'rootPath',
+		// from the acquired file name, 'file'
+		file = strings.TrimPrefix(file, rootPath)
```

Comment:
you would need to trim only `github.com/minio/minio` ?

Reply:
It is already trimming only `github.com/minio/minio`.
I achieved that by changing the `rootPath`. 

Please see `line#118`
`rootPath := filepath.Dir(filepath.Dir(file)) + string(os.PathSeparator)`

For example, 
`github.com/minio/minio/cmd/object-handlers.go`
shows up in stack trace as
`cmd/object-handlers.go:491:cmd.objectAPIHandlers.PutObjectHandler()`

Reply:
`filepath.Dir()` woudn't work as expected for gateway code e.g. when `file` is `github.com/minio/minio/cmd/gateway/azure/gateway-azure.go`, then `filepath.Dir(filepath.Dir(file))` returns `github.com/minio/minio/cmd/gateway` whereas, we would need to trim `github.com/minio/minio` always so that stacktrace would be like
```
main.go
cmd/gateway/azure/utils.go
cmd/utils.go
github.com/user/project/utils.go
...
```